src/main/java/com/gluonhq/substrate/target/AndroidTargetConfiguration.java

```diff
@@ -84,6 +95,8 @@ public AndroidTargetConfiguration( ProcessPaths paths, InternalProjectConfigurat
             this.ldlld = null;
             this.clang = null;
         }
+        this.java8Home = System.getenv("JAVA8_HOME");
```

Comment:
All the null checks below can be combined with variable retrieval like
`Objects.requireNotNull( System.getenv("var"), "Error message);`

Reply:
The problem is that those fields are not needed in the compile phase, hence failing on them in the constructor is not what we want.
They are needed in the link and run phase, so the assignments and checks can be in those 2 methods, at the price of code duplication. Not sure what is the worst.

Reply:
The reason for my comment was that you already throwing those exceptions just below, so why not to combine into one line.

Reply:
because they are not in the same method.